chiselFrontend/src/main/scala/chisel3/core/Data.scala

```diff
@@ -507,8 +507,8 @@ object Wire extends WireFactory
 object WireInit {
   def apply[T <: Data](init: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = {
     val model = (init match {
-      // For e.g. Wire(init=0.U(k.W)), fix the Reg's width to k
```

Comment:
Thanks for updating the comment.  One reason this was able to regress is that I only wrote the comment to cover the special case, and didn't explain the intended semantics for the other cases.